dask/dataframe/io/parquet/arrow.py

```diff
@@ -179,7 +179,7 @@ class PartitionObj:
 
     def __init__(self, name, keys):
         self.name = name
-        self.keys = sorted(keys)
+        self.keys = pd.Index(keys.sort_values())
```

Comment:
Using ``sort_values`` preserves the dtype (and is faster) and converting to an Index (no-copy op) enables us to use ``get_loc``